src/parallel/task_scheduler.cpp

```diff
@@ -144,8 +142,18 @@ void TaskScheduler::ExecuteForever(atomic<bool> *marker) {
 	shared_ptr<Task> task;
 	// loop until the marker is set to false
 	while (*marker) {
-		// wait for a signal with a timeout
-		queue->semaphore.wait();
+		if (allocator_background_threads) {
```

Comment:
Maybe `|| Allocator::SupportsFlush()` so the path below is not called outside of jemalloc?